components/events/Add.vue

```diff
@@ -1,91 +1,58 @@
 <script setup lang="ts">
-import { storeToRefs } from "pinia";
-import { getAllEvents, EventDifficulty } from "./utils";
+import { EventDifficulty } from "./utils";
 import type { EventType } from "./utils";
-import { useAuthStore } from "~/stores/auth";
 
-const showModal = ref(false);
-
-const { jwt, isLoggedIn } = storeToRefs(useAuthStore());
-
-const inputStartTime = ref("");
-const inputEndTime = ref("");
-
-const formData = ref<Omit<EventType, "id">>({
-  name: "",
-  location: "",
-  mazemapLink: "",
-  summary: "",
-  description: "",
-  slides: "",
-  image: "",
-  organizer: "",
-  unixStartTime: 0,
-  unixEndTime: 0,
-  difficulty: EventDifficulty.EASY, // Add the difficulty field to formData
+const props = defineProps<
+  | {
+      isEdit: false;
+      formValues?: formWithoutId;
+    }
+  | { formValues: EventType; isEdit: true }
+>();
+
+const emit = defineEmits<{
+  eventAdded: [values: formWithoutId];
+  eventEdited: [values: EventType];
+}>();
+
+type formType = Omit<EventType, "id"> & { id?: string };
+type formWithoutId = Omit<EventType, "id">;
+
+const formData = ref<formType>({
+  name: props?.formValues?.name ?? "",
+  location: props?.formValues?.location ?? "",
+  mazemapLink: props?.formValues?.mazemapLink ?? "",
+  summary: props?.formValues?.summary ?? "",
+  description: props?.formValues?.description ?? "",
+  slides: props?.formValues?.slides ?? "",
+  image: props?.formValues?.image ?? "",
+  organizer: props?.formValues?.organizer ?? "",
+  unixStartTime: props?.formValues?.unixStartTime ?? 0,
+  unixEndTime: props?.formValues?.unixEndTime ?? 0,
+  difficulty: props?.formValues?.difficulty ?? EventDifficulty.EASY, // Add the difficulty field to formData
 });
 
-const resetFormData = () => {
-  formData.value = {
-    name: "",
-    location: "",
-    mazemapLink: "",
-    summary: "",
-    description: "",
-    slides: "",
-    image: "",
-    organizer: "",
-    unixStartTime: 0,
-    unixEndTime: 0,
-    difficulty: EventDifficulty.EASY,
-  };
-};
+const inputStartTime = ref(convertToLocalDate(unixToDate(formData.value.unixStartTime)));
```

Comment:
why are we converting the dates to a local date?

Reply:
so we can hydrate the datetime-local pickers

Reply:
otherwise they're completely empty and the previous start and endtime get lost when editing an event (the Add.vue is being used for edits)

Reply:
does this shift timezones? if so, does it take into account that 4pm in June means 4pm BST, whilst 4pm in December is 4pm GMT?

Reply:
it works fine for UTC, i don't know about anything else. i'd just leave it like this, its admin only anyways

Reply:
we need to check this in case it screws up in BST: part-way through next term if we edit an event (without changing the time), it might starting shifting by an hour each edit. Maybe test it when you go home for Xmas?